bundles/org.openhab.io.mqttembeddedbroker/src/test/java/org/openhab/io/mqttembeddedbroker/internal/MqttEmbeddedBrokerServiceTest.java

```diff
@@ -175,9 +186,9 @@ public void testPersistence() throws InterruptedException, IOException, Executio
 
         // Stop server -> close persistence storage and sync it to disk
         subject.deactivate();
-
-        File jksFile = new File(subject.getPersistenceFilename());
         assertTrue(jksFile.exists());
+        // this is needed to ensure the file is correctly written
+        waitForAssert(() -> assertEquals(12288, jksFile.length()));
```

Comment:
isn't this platform dependent any other way to check?

Reply:
Not sure. It seems to work on windows and linux. In fact this is only an ugly workaround. There is no way (at least none that I know of) to check whether the broker completed its shutdown.  Before that is the case, the file is not completely written and the test will fail.